src/primitives/Avatar.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useMemo } from "react";
+import DefaultLogo from "@/assets/default_logo.png";
+import { match, P } from "ts-pattern";
+import { Avatar as ShadCNAvatar, AvatarFallback, AvatarImage } from "@/ui-shadcn/avatar";
+
+interface AvatarProps {
+  fallbackName?: string;
+  ipfsCID?: string;
+  url?: string;
+  size?: number;
+  ipfsBaseURL?: string;
+  defaultImage?: string;
+}
```

Comment:
```suggestion
interface AvatarProps {
  fallbackName?: string;
  src?: string;
  srcType?: "url" | "ipfs"
  baseUrl?: string;
  size?: number;
  defaultImage?: string;
}
```

What do you think about this way? It would be more scalable for future sources without changing the prop types isn't it?

And... I don't fully understand the fallbackName vs defaultImage...

if no ipfsCID and no url is passed as props, then the defaultImage is rendered
but if ipfsCID or url is defined... but not reachable, then the fallback is rendered.

wouldn't it be cleaner to set the fallback default to defaultImage, or am I missing something?

Reply:
And second point - ideally we never show the default image yeah, it's only really so broken data doesn't look like shit.  If a project or round has no images it will usually have a name, which is better to show than our default

Reply:
yeah - we have to have this logic somewhere and I'd rather have it once in the primitive instead of in the higher-level component (or the application).  Like someday soon we will have a "ProjectCard" where you pass in a project and we will pass everything from the Project down to this - this current approach means that the ProjectCard component won't have any complexity or logic at all related to the Avatar, it'll "just work".